### PR TITLE
Decouple input from update

### PR DESCRIPTION
**Pull request by threePhase**
Really rough version to decouple the keypresses from the `player:update` callback.

Still need to work on getting turning to actually turn when acceleration is not being directly applied.

### CODE REVIEW

**Comment by threePhase** [comment]
Tried adjusting the values we were using for our physics to something more "realistic", but this still does not work quite right.

File: src/entities/player.lua [hunk 1]
```diff
@@ -50,31 +71,48 @@ function player:update(ground, dt)
...
-  angle = self.body:getAngle()
   mass = self.body:getMass()
-  if love.keyboard.isDown("w") then
-    if love.keyboard.isDown("a") then
-      self.body:applyTorque(-self.turnMultiplier * inertia)
-    elseif love.keyboard.isDown("d") then
+
+  vx, vy = self.body:getLinearVelocity()
+  ax = (vx - self.vx) / dt
+  ay = (vy - self.vy) / dt
```
Comment (davisjc) [inline src/entities/player.lua]: These should be local along with some variables previously defined in this function.

File: src/entities/player.lua [hunk 2]
```diff
@@ -1,9 +1,11 @@
 local assets = require("assets")
 local player = {}
+local debug = true
 
 -- create car
 function player:load(width, height)
-  self.acceleration = 300
+  -- acceleration in m/s^2
+  self.acceleration = 6.5
```
Comment (davisjc) [inline src/entities/player.lua]: This is one step closer to where we oughta go with this.  I think when we read that article and look at this again we should think about a summation of forces.

File: src/entities/player.lua [hunk 3]
```diff
@@ -120,4 +155,48 @@ function rotateVector(x, y, angle)
...
+end
+
+function player:beginTurningLeft()
+  self.isturningleft = true
+  self.isturningright = false
+end
+
+function player:beginTurningRight()
+  self.isturningleft = false
+  self.isturningright = true
```
Comment (davisjc) [inline src/entities/player.lua]: If these handlers limit themselves to updating state about the control they represent, we could do things in the update loop such as going straight when both left and right are pressed, and thereafter if only left is released, we would know we could turn right.

**Review by davisjc — COMMENTED** [review]
Neat!

File: src/entities/player.lua [hunk 4]
```diff
@@ -41,6 +46,22 @@ function player:draw()
                      1,
                      self.width/2,
                      self.height/2)
+  if debug then
+    status = string.format("Mass: %.0f, Vx: %.2f, Vy: %.2f",
```
[the remark this answers is not in the document]
Comment (threePhase) [reply src/entities/player.lua]: Ah good call. I need to get this linting going so I stop missing that.

File: src/entities/player.lua (hunk 2 above)
Comment (threePhase) [reply src/entities/player.lua]: Yeah I initially started this work before I read that article, but I agree that this feels a lot more sane.

File: src/entities/player.lua (hunk 3 above)
Comment (threePhase) [reply src/entities/player.lua]: Good call, I'll adjust that.

**Review by davisjc — APPROVED** [review]
image

**Pull request merged**